sim/midas/src/main/scala/midas/passes/AutoCounterCoverTransform.scala

```diff
@@ -31,347 +31,207 @@ class FireSimPropertyLibrary extends BasePropertyLibrary {
     //1. Using an input file called `covermodules.txt` in a directory declared in the transform concstructor
     //2. Using chisel annotations to be added in the Platform Config (in SimConfigs.scala). The annotations are
     //   of the form AutoCounterModuleAnnotation("ModuleName")
     val modulesfile = new File(dir,"autocounter-covermodules.txt")
-    val filemoduleannos = mutable.ArrayBuffer.empty[AutoCounterCoverModuleFirrtlAnnotation]
+    val moduleAnnos     = new mutable.ArrayBuffer[AutoCounterCoverModuleFirrtlAnnotation]()
+    val counterAnnos    = new mutable.ArrayBuffer[AutoCounterFirrtlAnnotation]()
+    val remainingAnnos  = new mutable.ArrayBuffer[Annotation]()
     if (modulesfile.exists()) {
       val sourcefile = scala.io.Source.fromFile(modulesfile.getPath())
       val covermodulesnames = (for (line <- sourcefile.getLines()) yield line).toList
       sourcefile.close()
-      filemoduleannos ++= covermodulesnames.map {m: String => AutoCounterCoverModuleFirrtlAnnotation(ModuleTarget(state.circuit.main,m))}
+      moduleAnnos ++= covermodulesnames.map {m: String => AutoCounterCoverModuleFirrtlAnnotation(ModuleTarget(state.circuit.main,m))}
     }
-    val moduleannos = (state.annotations.collect {
-      case a: AutoCounterCoverModuleFirrtlAnnotation => a
-    } ++ filemoduleannos).distinct
-
-    //extract the module names from the methods mentioned previously
-    val covermodulesnames = moduleannos.map { case AutoCounterCoverModuleFirrtlAnnotation(ModuleTarget(_,m)) => m }
-
-    if (!covermodulesnames.isEmpty) {
-      println("[AutoCounter]: Cover modules in AutoCounterTransform:")
-      println(covermodulesnames)
+    state.annotations.foreach {
+      case a: AutoCounterCoverModuleFirrtlAnnotation => moduleAnnos += a
+      case a: AutoCounterFirrtlAnnotation => counterAnnos += a
+      case o => remainingAnnos += o
     }
 
-    //filter the cover annotations only by the modules that we want
-    val filtercoverannos = coverannos.filter{ case AutoCounterCoverAnnotation(ReferenceTarget(_,modname,_,_,_),l,m) =>
-                                           covermodulesnames.contains(modname) }
+    //extract the module names from the methods mentioned previously
+    val covermodulesnames = moduleAnnos.map(_.target.module).distinct
 
-    val allcounterannos = filtercoverannos ++ autocounterannos
-    //group the selected signal by modules, and attach label from the cover point to each signal
-    val selectedsignals = allcounterannos.map { case AutoCounterCoverAnnotation(target,l,m) => (target, l)
-                                                case AutoCounterFirrtlAnnotation(target,l,m) => (target, l)
-                                            }
-                                   .groupBy { case (ReferenceTarget(_,modname,_,_,_), l) => modname }
+    //collect annotations for manually annotated AutoCounter perf counters
+    val filteredCounterAnnos =  counterAnnos.filter(_.shouldBeIncluded(covermodulesnames))
```

Comment:
The only regression I see from the previous implementation is the following scenario:
I want to see the cover function counters from modules X, Y, Z + an additional couple of ad-hoc counter from module W